hat/app/org/hatdex/hat/utils/NamespaceUtils.scala

```diff
@@ -46,4 +47,55 @@ object NamespaceUtils {
     }
     matchedRoles.flatten.nonEmpty
   }
+
+  def getWriteNamespace(roles: Seq[UserRole]): Option[String] = {
+    val writeNamespaces = roles.map {
+      case NamespaceWrite(n) => n
+      case _                 => ""
+    }
+
+    if (writeNamespaces.isEmpty)
+      None
+    else
+      Some(writeNamespaces.head)
+  }
+
+  def getReadNamespace(roles: Seq[UserRole]): Option[String] = {
+    val readNamespaces = roles.map {
+      case NamespaceRead(n) => n
+      case _                => ""
+    }
+
+    if (readNamespaces.isEmpty)
+      None
+    else
+      Some(readNamespaces.head)
+  }
+
+  def verifyNamespace(
+      app: Application,
+      namespace: String): Boolean = {
+
+    val canReadNamespace  = verifyNamespaceRead(app, namespace)
+    val canWriteNamespace = verifyNamespaceWrite(app, namespace)
+
+    (canReadNamespace || canWriteNamespace)
```

Comment:
this is not correct, we need to know what operation is attempted (read or write?) and what the application is allowed to perform, then we can decide if it is allowed